pkg/controller/controller.go

```diff
@@ -1154,37 +1183,56 @@ func (p *csiProvisioner) Delete(ctx context.Context, volume *v1.PersistentVolume
 	req := csi.DeleteVolumeRequest{
 		VolumeId: volumeId,
 	}
-	// get secrets if StorageClass specifies it
```

Comment:
IMO the whole Secrets handling now deserves its own function, Delete() has grown too much.

Reply:
sure, moved into a seperate funcs.